### PR TITLE
Chai tests

### PR DESCRIPTION
**Pull request by herikudaru**
Worked out a set of tests for all the api functions, cleaned some things along the way.

### CODE REVIEW

File: test/test_api.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+    it('should modify a product', function(done) {
+        chai.request(process.env.TEST_API_ENDPOINT)
+        .put('')
+        .set('shared_secret_key', process.env.shared_secret_key)
+        .set('product_id', prodId)
+        .send({
+            "name": "Modified beer",
+            "description": "Ale, 3.1%",
+            "image": "example.com/12_new_final_final_new.png",
```
Comment (silentnoodlemaster) [inline test/test_api.js:120]: _nice_

File: test/test_api.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            expect(req).to.be.json;
+            expect(req.body).to.have.property('name');
+            expect(req.body).to.have.property('description');
+            expect(req.body).to.have.property('image');
+            expect(req.body).to.have.property('price');
+            done();
+        });
+    });
+
+    it('should delete a product', function(done) {
```
Comment (silentnoodlemaster) [inline test/test_api.js:134]: to confirm we could try to get a `404` with the `prodId`

Comment (silentnoodlemaster) [reply test/test_api.js:134]: discussed in person

**Review by silentnoodlemaster — APPROVED** [review]
:+1:

**Comment by MishClaw** [comment]
Even if merged, I read your changes, good job 👍

**Pull request merged**